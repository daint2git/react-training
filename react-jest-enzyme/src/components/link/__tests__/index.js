import Link from '../'

describe('Link', () => {
  test('using react-test-renderer', () => {
    const component = renderer.create(<Link url="http://www.facebook.com">Facebook</Link>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // manually trigger the callback
    tree.props.onMouseEnter()
    // re-rendering
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // manually trigger the callback
    tree.props.onMouseLeave()
    // re-rendering
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('using shallow', () => {
    const wrapper = shallow(<Link url="http://www.facebook.com">Facebook</Link>)

    expect(wrapper).toMatchSnapshot()
  })

  test('using mount', () => {
    const wrapper = mount(<Link url="http://www.facebook.com">Facebook</Link>)

    expect(wrapper).toMatchSnapshot()
  })

  test('using render', () => {
    const wrapper = render(<Link url="http://www.facebook.com">Facebook</Link>)

    expect(wrapper).toMatchSnapshot()
  })
})
