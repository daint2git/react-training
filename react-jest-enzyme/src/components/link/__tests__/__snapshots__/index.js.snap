// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link using mount 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link using react-test-renderer 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link using react-test-renderer 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link using react-test-renderer 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link using render 1`] = `
<a
  class="normal"
  href="http://www.facebook.com"
>
  Facebook
</a>
`;

exports[`Link using shallow 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
